api/discovery/discovery.go

```diff
@@ -51,7 +51,7 @@ type Authorizer interface {
 }
 
 // NewResource creates a new voc resource.
-func NewResource(d Discoverer, ID voc.ResourceID, name string, creationTime *time.Time, location voc.GeoLocation, labels map[string]string, typ []string) *voc.Resource {
+func NewResource(d Discoverer, ID voc.ResourceID, name string, creationTime *time.Time, location voc.GeoLocation, labels map[string]string, typ []string, raw string) *voc.Resource {
```

Comment:
Suggestion: Instead of directly specifying the raw string here, could we add a varargs any to the function like "rawObjects" and then use your voc.StringInterface function thingie to convert all the raw objects to a string and assign it here.

This would remove a lot of boiler-plate code out of the individual discovery functions. They would simply need to append the raw objects they want to consider when calling `NewResource`.